tools/taskcluster-worker-runner/provider/aws/awsprovider.go

```diff
@@ -53,6 +55,8 @@ func (p *AWSProvider) ConfigureRun(state *run.State) error {
 		return fmt.Errorf("Could not create worker manager client: %v", err)
 	}
 
+	p.workerManager = wm
```

Comment:
This is an un-authenticated client, so it won't be able to call RemoveWorker.  I think you'll need to re-invoke `p.workerManagerClientFactory` when shutting down, passing it the worker credentials.

Reply:
How can it call `RegisterWorker` then?

Reply:
With this many moving parts, I hardly think that qualifies as stupid :)

Reply:
Done